Source/WebCore/platform/mediastream/gstreamer/MockRealtimeAudioSourceGStreamer.h

```diff
@@ -34,7 +34,9 @@ class MockRealtimeAudioSourceGStreamer final : public MockRealtimeAudioSource {
 public:
     static Ref<MockRealtimeAudioSource> createForMockAudioCapturer(String&& deviceID, AtomString&& name, String&& hashSalt);
 
-    ~MockRealtimeAudioSourceGStreamer() = default;
+    static HashSet<MockRealtimeAudioSource*>& allMockRealtimeAudioSources();
```

Comment:
Ideally this would exposed a `const HashSet<MockRealtimeAudioSource*>&` since we don't expect anybody outside the class to modify the set.

Reply:
Right, good points! Thanks for the review! I guess I'll also apply these changes on the MockRealtimeVideoSource side as well.

Reply:
> Ideally this would exposed a `const HashSet<MockRealtimeAudioSource*>&` since we don't expect anybody outside the class to modify the set.

Hm I tried this but then compiler was choking on `allMockRealtimeAudioSources().add(this);` and remove call, called from the ctor and dtor respectively.

Reply:
Right, so the usual pattern is to have a static function in the cpp that returns a non-const reference, that you use from your constructor & destructor. Then have a public static function in the header that returns a const reference (and which calls the internal cpp function internally).